plugins/npm/nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/service/internal/orient/OrientMetadataStore.java

```diff
@@ -321,40 +322,48 @@ public int updatePackages(final NpmRepository repository, final Iterator<Package
   }
 
   @Override
-  public int updatePackages(final NpmRepository repository, final Predicate<PackageRoot> predicate,
+  public int updatePackages(final NpmRepository repository,
+                            final Predicate<PackageRoot> predicate,
                             final Function<PackageRoot, PackageRoot> function)
   {
     checkNotNull(repository);
     checkNotNull(function);
     final EntityHandler<PackageRoot> entityHandler = getHandlerFor(PackageRoot.class);
+    int count = 0;
     try (ODatabaseDocumentTx db = db()) {
       db.begin();
-      try {
-        final OSQLSynchQuery<ODocument> query = new OSQLSynchQuery<>(
-            "select from " + entityHandler.getSchemaName() + " where repositoryId='" + repository.getId() +
-                "' and @rid > ? limit 1000");
-        ORID last = new ORecordId();
-        List<ODocument> resultset = db.query(query, last);
-        int count = 0;
-        while (!resultset.isEmpty()) {
-          for (ODocument doc : resultset) {
-            PackageRoot root = entityHandler.toEntity(doc);
+      final OSQLSynchQuery<ODocument> query = new OSQLSynchQuery<>(
+          "select @rid as orid from " + entityHandler.getSchemaName() + " where repositoryId='" + repository.getId() +
+              "' and @rid > ? limit 1000");
+      ORID last = new ORecordId();
+      List<ODocument> resultset = db.query(query, last);
+      while (!resultset.isEmpty()) {
+        try {
+          for (ODocument d : resultset) {
+            final ORID orid = d.field("orid", ORID.class);
+            final ODocument npmDoc = db.load(orid);
+            if (npmDoc == null) {
+              continue;
+            }
+            PackageRoot root = entityHandler.toEntity(npmDoc);
             if (predicate != null && !predicate.apply(root)) {
               continue;
             }
             root = function.apply(root);
-            db.save(entityHandler.toDocument(root, doc));
+            db.save(entityHandler.toDocument(root, npmDoc));
             count++;
           }
-          last = resultset.get(resultset.size() - 1).getIdentity();
+          db.commit();
+          last = resultset.get(resultset.size() - 1).field("orid", ORID.class);
```

Comment:
we could keep track of `last` inside the for loop as we now have the `orid` available as we work through the docs